sdk/framework/openapi.go

```diff
@@ -534,31 +530,624 @@ func documentPath(p *Path, specialPaths *logical.Paths, requestResponsePrefix st
 	return nil
 }
 
-// constructRequestResponseName joins the given path with prefix & suffix into
-// a CamelCase request or response name.
+// When constructing a request or response name from a prefix + path, certain
+// paths result in very long names or names with duplicate parts. For such paths,
+// we use custom path mappings instead to generate request/response names.
+type knownPathKey struct {
+	mount string
+	path  string
+}
+
+type knownPathValue struct {
+	name             string
+	operationImplied bool // if true, operation will not be appended to the resulting name
+}
+
+var knownPathMappings = map[knownPathKey]knownPathValue{
+	{mount: "auth/alicloud", path: "login"}:                                            {name: "AliCloudLogin", operationImplied: true},
+	{mount: "auth/alicloud", path: "role"}:                                             {name: "AliCloudAuthRoles"},
```

Comment:
Is this how the "default mount path" comes through? `auth/alicloud` vs. just `alicloud`? I think it's a little bit of a misnomer b/c I don't think `auth` is considered part of the mount

Reply:
Yes, this is how defaultMountPath comes through. There is an entry for "auth/alicloud" (corresponding to the auth backend) and "alicloud" (corresponding to the secrets backend). The "auth" part is also referred to as "mountPrefix". `defaultMountPath` might not be a good name. Do you have any suggestions?

Reply:
Ahh. I see. I think it's fine if that's how it comes in. It may just be adding more complexity to split it out anyway.

Reply:
renamed this to `defaultMountPath` to `mountPathWithPrefix` to avoid confusion.